homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,232 @@
+                    self._current_swing_mode = value.data
+                    self._swing_list = [0, 1]
+                    _LOGGER.debug('self._swing_list=%s)', self._swing_list)
+
+        else:
+            for _, value in self._node.get_values(class_id=0x42).items():
+                self._current_operation_state = value.data
+                self._fan_list = list(value.data_items)
+                _LOGGER.debug('self._fan_list=%s)', self._fan_list)
+
+    @property
+    def should_poll(self):
+        """No polling on ZWave."""
+        return False
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan speed set."""
+        return self._current_operation_state
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    @property
+    def current_swing_mode(self):
+        """Return the swing mode set."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        unit = self._unit
+        if unit == 'C':
+            return TEMP_CELSIUS
+        elif unit == 'F':
+            return TEMP_FAHRENHEIT
+        else:
+            return unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return self._operation_list
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, temperature):
+        """Set new target temperature."""
+        # ZXT-120 does not support get setpoint
+        self._target_temperature = temperature
+        for value in self._node.get_values(class_id=0x43).values():
+            if value.command_class != 67:
+                continue
+            if self._zxt_120 == 1:
+                # pylint: disable=too-many-boolean-expressions
+                if self._current_operation == 'Heat' and value.index == 1 or \
+                   self._current_operation == 'Cool' and value.index == 2 or \
+                   self._current_operation == 'Dry Air' and \
+                   value.index == 8 or \
+                   self._current_operation == 'Auto Changeover' and \
+                   value.index == 10:
+                    # ZXT-120 responds only to whole int
+                    value.data = int(round(temperature, 0))
+            else:
+                for value in self._node.get_values(class_id=0x43).values():
```

Comment:
You can remove this line since you already iterate exactly the same way on line 182.